### DISCUSSION TITLE
Problems with the installation

### DISCUSSION THREAD

**Discussion by Johnny0514** [LAPKB/Pmetrics#208] [Q&A]
I have tried to install devtools in R studio, but I failed.
The problems  were suggested as below:
Error in loadNamespace(j <- i[[1L]], c(lib.loc, .libPaths()), versionCheck = vI[[j]]) : 
  namespace 'cli' 2.5.0 is being loaded, but >= 3.1.0 is required
Calls: <Anonymous> ... namespaceImportFrom -> asNamespace -> loadNamespace
停止执行
ERROR: lazy loading failed for package 'sessioninfo'
* removing 'C:/Users/LENOVO/Documents/R/win-library/3.6/sessioninfo'
Warning in install.packages :
  installation of package ‘sessioninfo’ had non-zero exit status
ERROR: dependency 'lifecycle' is not available for package 'httr2'
* removing 'C:/Users/LENOVO/Documents/R/win-library/3.6/httr2'
Warning in install.packages :
  installation of package ‘httr2’ had non-zero exit status
ERROR: dependency 'lifecycle' is not available for package 'bslib'
* removing 'C:/Users/LENOVO/Documents/R/win-library/3.6/bslib'
Warning in install.packages :
  installation of package ‘bslib’ had non-zero exit status
ERROR: dependency 'lifecycle' is not available for package 'stringr'
* removing 'C:/Users/LENOVO/Documents/R/win-library/3.6/stringr'
Warning in install.packages :
  installation of package ‘stringr’ had non-zero exit status
ERROR: dependency 'callr' is not available for package 'pkgbuild'
* removing 'C:/Users/LENOVO/Documents/R/win-library/3.6/pkgbuild'
Warning in install.packages :
  installation of package ‘pkgbuild’ had non-zero exit status
ERROR: dependency 'httr2' is not available for package 'gh'
* removing 'C:/Users/LENOVO/Documents/R/win-library/3.6/gh'
Warning in install.packages :
  installation of package ‘gh’ had non-zero exit status
ERROR: dependencies 'bslib', 'knitr', 'stringr', 'tinytex' are not available for package 'rmarkdown'
* removing 'C:/Users/LENOVO/Documents/R/win-library/3.6/rmarkdown'
Warning in install.packages :
  installation of package ‘rmarkdown’ had non-zero exit status
ERROR: dependency 'pkgbuild' is not available for package 'pkgload'
* removing 'C:/Users/LENOVO/Documents/R/win-library/3.6/pkgload'
Warning in install.packages :
  installation of package ‘pkgload’ had non-zero exit status
ERROR: dependencies 'callr', 'pkgbuild', 'sessioninfo' are not available for package 'rcmdcheck'
* removing 'C:/Users/LENOVO/Documents/R/win-library/3.6/rcmdcheck'
Warning in install.packages :
  installation of package ‘rcmdcheck’ had non-zero exit status
ERROR: dependencies 'gh', 'lifecycle' are not available for package 'usethis'
* removing 'C:/Users/LENOVO/Documents/R/win-library/3.6/usethis'
Warning in install.packages :
  installation of package ‘usethis’ had non-zero exit status
ERROR: dependencies 'bslib', 'callr', 'rmarkdown' are not available for package 'pkgdown'
* removing 'C:/Users/LENOVO/Documents/R/win-library/3.6/pkgdown'
Warning in install.packages :
  installation of package ‘pkgdown’ had non-zero exit status
ERROR: dependencies 'usethis', 'lifecycle', 'pkgbuild', 'pkgdown', 'pkgload', 'rcmdcheck', 'sessioninfo' are not available for package 'devtools'
* removing 'C:/Users/LENOVO/Documents/R/win-library/3.6/devtools'
Warning in install.packages :
  installation of package ‘devtools’ had non-zero exit status
What should I do next?

**Comment by masyamada** [comment]
Have you tried to reinstall the cli package?  I am using version 3.6.2.

If you are using Rstudio: go to your package pane window and click the
green circle (updates). It should prompt you to update CLI and any other
packages that have a more recent public version.

"namespace 'cli' 2.5.0 is being loaded, but >= 3.1.0 is required"

-wy.

**Reply by mnneely** [reply]
We've been finding that packages which Pmetrics depends on may have specific version requirements of their own dependencies (ie. package grandchildren of Pmetrics). This can cause failure of automatic installation of dependencies. Try updating all packages in your R installation (the Update button in the Packages tab in Rstudio) prior to installing Pmetrics.
<img width="162" alt="image" src="https://github.com/LAPKB/Pmetrics/assets/8136142/2c2cefd3-c5ad-4d89-a9dc-770cd15c9b81">
